tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -3802,7 +3780,15 @@ Status TfTensorToTrtWeights(const Tensor& tensor, TrtWeightStore* weight_store,
   }
 
   nvinfer1::Dims weight_dims;
-  GetTensorDimsWithProtoShape(tensor, &weight_dims);
+  if (tensor.dims() > 0) {
+    weight_dims = TensorShapeToTrtDimsUnsafe(tensor.shape());
+  } else {
+    // Use scalar weights to implement scalar constants.
+    // No dimension provided. Flatten it.
+    weight_dims.nbDims = 0;
+    weight_dims.d[0] = static_cast<int32_t>(tensor.NumElements());
```

Comment:
I understand this is inlined from the existing code. But will the only possible value for tensor.NumElements() is 1 here? Will it be easier to understand if we return error if it is not 1?

Reply:
Fixed by simplifying surrounding code block

Reply:
We were using it in TRT_ShapedWeights default constructor implicitly. I removed it and made TRT_ShapedWeights default constructor explicitly initialize the shape to the the zero shape.